src/lib/core/PeerId.h

```diff
@@ -39,19 +41,53 @@ class PeerId
         return *this;
     }
 
-    FabricId GetFabricId() const { return mFabricId; }
-    PeerId & SetFabricId(FabricId id)
+    CompressedFabricId GetCompressedFabricId() const { return mCompressedFabricId; }
+    PeerId & SetCompressedFabricId(CompressedFabricId id)
+    {
+        mCompressedFabricId = id;
+        return *this;
+    }
+
+    bool operator==(const PeerId & other) const
+    {
+        return (mNodeId == other.mNodeId) && (mCompressedFabricId == other.mCompressedFabricId);
+    }
+    bool operator!=(const PeerId & other) const
+    {
+        return (mNodeId != other.mNodeId) || (mCompressedFabricId != other.mCompressedFabricId);
+    }
+
+private:
+    NodeId mNodeId = kUndefinedNodeId;
+
+    CompressedFabricId mCompressedFabricId = kUndefinedFabricId;
+};
+
+class RawPeerId
```

Comment:
Why do we need this?  This is not a raw node identifier; it's an incomplete node identifier.  If we do need it, can we have a TODO and a ticket?  This needs to go away.

Also, suggest a name like `IncompleteNodeIdentifier` or similar so nobody mistakes this for a guaranteed-unique key; it's not.